linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -404,7 +405,7 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
         case Transient(JSVarRef(ident, _)) => ident
         case record: RecordSelect          => makeRecordFieldIdent(record)
       }
-      makeRecordFieldIdent(recIdent, tree.field.name, tree.field.originalName)
+      makeRecordFieldIdent(recIdent, tree.field.name, NoOriginalName)
```

Comment:
Ugh, this is not nice... So depending on how we construct the full record field identifier we are able to retain the original name or not :(